source/core/types/ut_run.tpb

```diff
@@ -75,6 +75,14 @@ create or replace type body ut_run as
     return l_completed_without_errors;
   end;
 
+  overriding member procedure set_rollback_type(self in out nocopy ut_run, a_rollback_type integer, a_force boolean := false) is
+  begin
+    self.rollback_type := case when a_force then a_rollback_type else coalesce(self.rollback_type, a_rollback_type) end;
+    for i in 1 .. self.items.count loop
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Replace this FOR loop by a more robust WHILE one. ([plsql:ForLoopUsageCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3AForLoopUsageCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=789&issues=AWckVbzqd6oxh3R4GXWS&open=AWckVbzqd6oxh3R4GXWS)

[ISSUE_KEY]: AWckVbzqd6oxh3R4GXWS